### ISSUE TITLE
Prompt to enable Bluetooth when it is disabled

### ISSUE THREAD

**Issue by AtmosphericIgnition** [mjaakko/NeoStumbler#270]
### Description

I do not use any Bluetooth devices on a regular basis, but I do scan for Bluetooth beacons through NeoStumbler. When I forget to turn on Bluetooth, and start scanning with NeoStumbler, it just collects data without the Bluetooth beacons. I know that an Android app can ask the user to enable Bluetooth when it is disabled through the this prompt:

![signal-2024-10-05-184042](https://github.com/user-attachments/assets/a772b6f1-6291-413a-964f-d724335cdcc6)

### Suggested implementation

When the user starts scanning without Bluetooth enabled on their device, they should be prompted to enable it.

### Additional information

https://developer.android.com/develop/connectivity/bluetooth/setup

### New request

- [X] #305

**Comment by mjaakko** [comment]
I'm thinking about UX for this

* If you don't want to scan Bluetooth beacons for some reason and don't enable Bluetooth, should it prompt to enable Bluetooth the next time you start scanning?
* If you start scanning from the quick settings tile,  would it be better to open the UI for the prompt or to just start scanning without Bluetooth?

As a workaround, see if you can enable Bluetooth scanning to be always available in your device settings:
![Screenshot_20241102_130710_Settings](https://github.com/user-attachments/assets/dda37cf0-c1f6-4270-a43c-2057b9d2d39e)

**Comment by skinkie** [comment]
This technically also applies to "Location" in general. If location is disabled, there is no indication why scanning does not start.

**Comment by lgommans** [comment]
And also for WiFi please! I recently forgot to enable that and did some kilometres where I'd probably never go again, only to notice I didn't hit the system button correctly x_x

For the UI, I would suggest a notice on the page that appears when you start scanning and one of the options/dependencies (location, bluetooth, wifi, no flight mode) is not enabled. A pop-up as shown in the OP usually just annoys me. Each optional 'dependency' could have two buttons, one for "don't show this again" and one for just "dismiss for this session / today", so that people can decide if they just never want to scan one of these types or just not now

<img width="408" height="380" alt="Mock-up of the current reports screen, having added in a notice just above the 'start scanning' button with an error symbol and the text 'Location disabled, scanning not possible', and two warning symbols before two messages 'WiFi disabled' and 'Bluetooth disabled'. The latter two have two buttons, one crossed-out eye and one x to do not show again and just dismiss for now, respectively. These would have alt and title/tooltip texts, I imagine" src="https://github.com/user-attachments/assets/d35cf9d3-f12d-4b8d-8f4a-d669ed2ecae2" />

Not sure if there are other ways to enable scanning, like from a tile in the system controls area iirc? I don't use those; there, we might need that text to be shown in a pop-up instead

**Comment by Yaya-Cout** [comment]
I've also encountered a similar problem: I often have my phone in airplane mode with Bluetooth and Wi-Fi disabled, and launch Neostumbler from the quick tile. In this case, Neostumbler can't scan anything, but doesn't show any warning. 
I would suggest adding a notification to warn the user when scanning isn't possible.

**Comment by mjaakko** [comment]
> And also for WiFi please! 

For Wi-Fi, this is not possible, because Android does not provide an API to request enabling Wi-Fi. But you should be able to do scans with Wi-Fi off if you have enabled "Wi-Fi scan always available". From NeoStumbler settings -> Troubleshooting you can try to enable this. Unfortunately (as usual..) it does not work reliably on all devices and some devices might report that they can do Wi-Fi scans when in reality it does not work.

> Each optional 'dependency' could have two buttons, one for "don't show this again" and one for just "dismiss for this session / today", so that people can decide if they just never want to scan one of these types or just not now

IMO, having a "never show again" button is not good UX on mobile devices, because it's too easy to accidentally tap it and then there's no way to reset it. I think the UI could be improved here, e.g. split report list and scanning dashboard to separate tabs and then there would be more space to show these types of warnings

**Comment by lgommans** [comment]
Yeah I agree that "never show again" (the eye icon in my mock-up) shouldn't result in a permanent change upon first touch. Someone might just be trying out what it does, or tap it accidentally as you said. It could:

- trigger a pop-up "Should the Bluetooth setting reminder be **permanently** hidden? <kbd>Hide forever</kbd> <kbd>Hide once</kbd> <kbd>Cancel</kbd>"
- be a menu instead (e.g.: three vertical dots), opening up vertically above or below the menu button and having just this option, so you need to do two taps in different spots to effect this change.

The settings screen could have a "Warnings" or "Reminders" section, with a toggle for the three warning/reminder types (WiFi, BT, and cell). It would indeed be silly if there is no way to change your preference short of deleting all app data, but probably it's even more annoying to get warnings or pop-ups of something you don't want to collect (I can imagine a user not wanting to collect Bluetooth beacons, where giving them the warning over and over is just a pain point every time)

> For Wi-Fi, this is not possible, because Android does not provide an API to request enabling Wi-Fi.

I wouldn't want it to! Also not for Bluetooth or cells. Not only does it feel out of scope for a user-level app to prompt about changing a system or device setting, it's also annoying to get a pop-up when you maybe did this on purpose and are trying to use the rest of the UI still.

Just a visual confirmation that it started scanning after I press 'scan', or a reminder in its place if not, would do it for me personally
